chat/src/main/java/com/github/twitch4j/chat/events/channel/DirectCheerEvent.java

```diff
@@ -0,0 +1,137 @@
+package com.github.twitch4j.chat.events.channel;
+
+import com.github.twitch4j.common.annotation.Unofficial;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.Currency;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This event gets called when a user does a direct cheer in an eligible channel for this experiment.
+ *
+ * @see <a href="https://help.twitch.tv/s/article/cheering-experiment-2022">Twitch Information</a>
+ */
+@Value
+@Unofficial
+@EqualsAndHashCode(callSuper = true)
+public class DirectCheerEvent extends CheerEvent {
+
+    private static final Map<String, Currency> CURRENCY_MAP;
+
+    /**
+     * Creator receives 80% of the amount after fees, during the experiment.
+     *
+     * @see <a href="https://help.twitch.tv/s/article/cheering-experiment-2022?language=en_US#StreamerFAQ">Help Article</a>
+     */
+    private static final float CREATOR_REVENUE_SPLIT = 0.8F;
+
+    /**
+     * The amount of the direct cheer.
+     * <p>
+     * For example, $1 = 100
+     * <p>
+     * Note: this is before adjustment for twitch's revenue split.
+     * Use {@link #getBits()} for the adjusted value.
+     */
+    Integer amount;
+
+    /**
+     * The twitch notification text for this event.
+     * <p>
+     * For example: DisplayName Cheered with $50
+     * <p>
+     * Note: this is distinct from the user's message attached to the cheer ({@link #getMessage()})
+     */
+    @NotNull
+    String systemMessage;
+
+    /**
+     * Parses the currency used in this direct cheer, or null if unknown.
+     * Currently, this always resolves to USD due to the experiment restrictions.
+     */
+    @Nullable
+    @Getter(lazy = true)
+    Currency currency = parseCurrency(getSystemMessage());
+
+    /**
+     * Parses the monetary value that was directly cheered.
+     * <p>
+     * For example, $50 is parsed to 50
+     */
+    @Nullable
+    @Getter(lazy = true)
+    BigDecimal monetaryValue = parseValue(getSystemMessage());
+
+    /**
+     * Event Constructor
+     *
+     * @param event The raw message event.
+     */
+    public DirectCheerEvent(IRCMessageEvent event) {
+        this(
+            event,
+            event.getTagValue("msg-param-amount").map(Integer::parseInt).orElse(null),
+            event.getTagValue("system-msg").map(String::trim).orElse("")
+        );
+    }
+
+    DirectCheerEvent(IRCMessageEvent event, Integer amount, @NotNull String systemMessage) {
+        super(event, event.getChannel(), event.getUser(), event.getMessage().orElse(""), Math.round(amount * CREATOR_REVENUE_SPLIT), event.getSubscriberMonths().orElse(0), event.getSubscriptionTier().orElse(0), event.getFlags());
```

Comment:
Just to be save, should we make sure that `amount` arg can not be null because of `Math.round(amount * CREATOR_REVENUE_SPLIT)` ? Just in case they change sth. up as it's still expermental.

Maybe by checking that `msg-param-amount` is present on not null in IRCEventHandler#onDirectCheer? (or default to 0 instead of null in line 81)

Reply:
good point, would agree with defaulting to `0` or `-1` (maybe `0` is best in case people are tracking tips over time & forget to check sign & tip amount falsely decreases marginally)